pyiceberg/catalog/glue.py

```diff
@@ -93,6 +93,13 @@
 if TYPE_CHECKING:
     import pyarrow as pa
 
+
+# There is a unique Glue metastore in each AWS account and each AWS region. By default, GlueCatalog chooses the Glue
+# metastore to use based on the user's default AWS client credential and region setup. You can specify the Glue catalog
+# ID through glue.id catalog property to point to a Glue catalog in a different AWS account. The Glue catalog ID is your
+# numeric AWS account ID.
```

Comment:
Documentation cribbed from [the Iceberg docs](https://iceberg.apache.org/docs/1.4.0/aws/#glue-catalog-id).